File: 107/107.c
#include <stdio.h>
#include <stdlib.h>
#include <math.h>
#include <tgmath.h>

double summationd(double a, double r, double n)
{
    return a * (1.0 - pow(r, n)) / (1.0 - r);
}

long long summation(long long a, long long num, long long dem, long long n)
{
    double r;

    if (dem == num)
        return a * n;

    r = ((double) num) / ((double) dem);

    return llrint(summationd(a, r, n));
}

int main(int argc, char **argv)
{
    char line[256];
    long long a, b;
    long long r0, r1;
    long long N, n;
    long long l, r;
    double c, d;

    do {
        fgets(line, 256, stdin);
        sscanf(line, "%lld %lld", &a, &b);

        if (a == 0 && b == 0) {
            break;
        }

        if (b == 1) {
            N = 1;
        } else {
            c = log(a) / log(b);

            d = -1.0;

            l = 2;
            r = a;

            do {
                N = (l + r) / 2;

                if (l == N)
                    break;

                d = log(N + 1) / log(N);

                if (d < c)
                    r = N;
                else
                    l = N;
            } while (1);
        }

        n = llrint(1.0 + log(a) / log(N + 1));

        r0 = summation(1, N, 1, n - 1);
        r1 = summation(a, N, N + 1, n);

        printf("%lld %lld\n", r0, r1);
    } while (1);

    return EXIT_SUCCESS;
}

File: 107/107.py
from math import log

def summation(a, r, n):
    if r != 1.0:
        return a * (r ** n - 1) / (r - 1)
    else:
        return a * n

while True:
    line = input().split()

    a = int(line[0])
    b = int(line[1])

    if a == 0 and b == 0:
        print('0 0')
        break

    stop = False

    if b == 1:
        if a == 1:
            print('0 1')
            continue
        else:
            N = 1
    else:
        for N in range(2, a):
            if abs(log(a, b) - log(N + 1, N)) < 0.000001:
                break

    n = int(round(log(a, N + 1)))

    r_a = summation(1, N, n)
    r_b = summation(a, float(N) / float(N + 1), n + 1)

    s = '{} {}'.format(int(r_a), int(round(r_b)))

    print(s)
